protoc-gen-twirp_php/templates/service/Server.php.tmpl

```diff
@@ -335,7 +335,7 @@ final class {{ .Service | phpServiceName .File }}Server implements RequestHandle
             'code' => $e->getErrorCode(),
             'msg' => $e->getMessage(),
             'meta' => $e->getMetaMap(),
-        ]));
+        ], \JSON_FORCE_OBJECT));
```

Comment:
Hey @sagikazarmark although this looks much better now, it still doesn't adhere to the spec https://twitchtv.github.io/twirp/docs/errors.html

> meta: (optional) key-value pairs with arbitrary string metadata. Useful to define subtypes under the same code, or add extra fields for the callers.

The `meta` key should be _optional_. So it should be omitted if it equals to the empty object.

Reply:
Last commit should fix that too

Reply:
I just quickly looked what Twitch's Go impl does, and indeed they're making it optional https://github.com/twitchtv/twirp/blob/4e1d44e644d7399417a9c9273dca725d98e731dd/errors.go#L102

Removing `meta` entirely could probably introduce BC breaks. Although, it's a bug fix to conform to the spec.
Perhaps the safest way is to deprecate the old behavior, bump the version, in the next major version fully remove the key if it's empty.

Not sure what's your release process here guys and if you're ok going straight to BC break.

Reply:
> Removing `meta` entirely could probably introduce BC breaks.

I don't think that's a concern here. The important thing is that clients can interpret what's traveling over the wire.

From what I can tell though, an empty map is going to be virtually the same as not sending it.

Optional in the Go implementation means that it's either not sent at all, or it's nil. Once again, that's virtually the same thing and even if clients are unable to handle the difference, spec conformity tilts the scale to the direction of including this fix sooner than later.